python/gilded_rose.py

```diff
@@ -6,34 +6,43 @@ def __init__(self, items):
         self.items = items
 
     def update_quality(self):
+
         for item in self.items:
-            if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert":
-                if item.quality > 0:
-                    if item.name != "Sulfuras, Hand of Ragnaros":
-                        item.quality = item.quality - 1
+
+            if item.name == "Aged Brie":
+                item.quality += 1
+
+            elif item.name.startswith("Backstage passes"):
+                if item.sell_in > 10:
+                    item.quality += 1
+                elif 10 >= item.sell_in > 5:
+                    item.quality += 2
+                elif 5 >= item.sell_in > 0:
+                    item.quality += 3
+                else:
+                    item.quality = 0
+
             else:
-                if item.quality < 50:
-                    item.quality = item.quality + 1
-                    if item.name == "Backstage passes to a TAFKAL80ETC concert":
-                        if item.sell_in < 11:
-                            if item.quality < 50:
-                                item.quality = item.quality + 1
-                        if item.sell_in < 6:
-                            if item.quality < 50:
-                                item.quality = item.quality + 1
-            if item.name != "Sulfuras, Hand of Ragnaros":
-                item.sell_in = item.sell_in - 1
-            if item.sell_in < 0:
-                if item.name != "Aged Brie":
-                    if item.name != "Backstage passes to a TAFKAL80ETC concert":
-                        if item.quality > 0:
-                            if item.name != "Sulfuras, Hand of Ragnaros":
-                                item.quality = item.quality - 1
+                if item.name != "Sulfuras, Hand of Ragnaros":
+                    if item.quality > 0:
+                        if item.sell_in > 0:
+                            if item.name.startswith("Conjured"):
+                                item.quality -= 2
+                            else:
+                                item.quality -= 1
+                        else:
+                            if item.name.startswith("Conjured"):
+                                item.quality -= 4
+                            else:
+                                item.quality -= 2
                     else:
-                        item.quality = item.quality - item.quality
-                else:
-                    if item.quality < 50:
-                        item.quality = item.quality + 1
+                        item.quality = 0
+
+            if item.quality > 50 and item.name != "Sulfuras, Hand of Ragnaros":
+                item.quality = 50
+
+            if item.name != "Sulfuras, Hand of Ragnaros":
```

Comment:
Considering `Sulfuras` is repeating 3 times, maybe we can check if the item is `Sulfuras` at the beginning and skip that step: hint `continue`